topdown/crypto.go

```diff
@@ -447,6 +472,32 @@ func readCertFromFile(localCertFile string) ([]byte, error) {
 	return certPEM, nil
 }
 
+func getTLSx509KeyPairFromString(certPemBlock []byte, keyPemBlock []byte) (*tls.Certificate, error) {
+
+	if !strings.HasPrefix(string(certPemBlock), "-----BEGIN") {
+		s, err := base64.StdEncoding.DecodeString(string(certPemBlock))
+		if err != nil {
+			return nil, err
+		}
+		certPemBlock = s
+	}
+
+	if !strings.HasPrefix(string(keyPemBlock), "-----BEGIN") {
+		s, err := base64.StdEncoding.DecodeString(string(keyPemBlock))
+		if err != nil {
+			return nil, err
+		}
+		keyPemBlock = s
```

Comment:
suggestion: in order to be consistent with the implementation of `builtinCryptoX509ParseAndVerifyCertificates` we might want to also handle the case where the data is actually a PEM block but passed as a base64 encoded string. See `getX509CertsFromString`

Reply:
I might've misunderstood what you mean, but I thought that was what I was doing here?

```
if !strings.HasPrefix(string(certPemBlock), "-----BEGIN") {
		s, err := base64.StdEncoding.DecodeString(string(certPemBlock))
		if err != nil {
			return nil, err
		}
		certPemBlock = s
	}

	if !strings.HasPrefix(string(keyPemBlock), "-----BEGIN") {
		s, err := base64.StdEncoding.DecodeString(string(keyPemBlock))
		if err != nil {
			return nil, err
		}
		keyPemBlock = s
	}
	
	cert, err := tls.X509KeyPair(certPemBlock, keyPemBlock)
	if err != nil {
		return nil, nil
	}
```
i.e if they do not match PEM assume its b64 and convert it, else just load them.. Or am I missing something? :)

Reply:
Ahh, I got confused. I should have said "might want to also handle the case where the data is actually **some DER bytes** but passed as a base64 encoded string".

That would be this case in the current implementations: https://github.com/charlieegan3/opa/blob/main/topdown/crypto.go#L304

In summary, I think there are three cases we need to handle:
* [x] non base64 encoded PEM block
* [x] base64 encoded PEM block
* [ ] base64 encoded DER data

Hope that makes sense. I'm not sure how well used the DER data path is in the existing functions (I think it'd be uncommon), however I think it might be worth supported in order to keep it consistent with the other functions.

Reply:
the logic is ok, but the underlying tls/crypto function does not support DER.

Reply:
@charlieegan3  see fb4b276b5415ba458edba6559e41c66e16ea909c.

Reply:
That looks good to me, thanks for adding this 👍